### PR TITLE
Policy compliance data enable pagination

### PR DESCRIPTION
**Pull request by najagasi**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories.

### CODE REVIEW

**Comment by raych1** [comment]
@najagasi , please fix the check failures of avocado. There're many un-referenced example files needs to be removed.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json]: >"swagger": "2.0", [](start = 2, length = 17)

Please remove PolicyEvents_GetMetadata.json, as it's not referenced anywhere.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyStates.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyStates.json]: readme.md is not referencing 2018-07-01-preview tag, so this change is not really going to be effective. Either revert this json or add 2018-07-01-preview tag to reference this spec file.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json]: >2019-10-01 [](start = 16, length = 10)

I think it is not yet implemented. I suggest move this file under 2018-07-01-preview. And add here once policyEvents is implemented for this version.

Comment (sandipsh) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json]: rather better, we could skip adding it to preview and directly move to latest version for policyEvents. 

---
In reply to: [355050603](https://github.com/Azure/azure-rest-api-specs/pull/7909#discussion_r355050603) [](ancestors = 355050603)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyStates.json
Comment (sandipsh) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyStates.json]: never mind, i noticed it's used for 2018-07 tag,

---
In reply to: [355047766](https://github.com/Azure/azure-rest-api-specs/pull/7909#discussion_r355047766) [](ancestors = 355047766)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (najagasi) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json]: had added to 2019-10 as you had suggested, will move it to 2018-07-01-preview since its not implemented yet.

**Comment by najagasi** [comment]
Please do not merge this PR yet

**Comment by raych1** [comment]
> Please do not merge this PR yet

Hi @najagasi , are you still working on this PR? Can you please close this one if not?

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:99]: >?api-version=2019-10-01&$skipToken=WpmWfBSvPhkAK6QD [](start = 120, length = 51)

move to parameters.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:103]: >parameters [](start = 9, length = 10)

also add skiptoken.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:37]: >parameters [](start = 9, length = 10)

add skiptoken. same everywhere else.

**Comment by sandipsh** [comment]
>     "title": "PolicyStatesClient",

Do you need to add PolicyEventsClient for this version? 

---
Refers to: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyStates.json:4 in 03198ee. [](commit_id = 03198eed76a3e7742052629e820ed005fd63a234, deletion_comment = False)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (najagasi) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:103]: thanks a lot, i had missed that

**Comment by najagasi** [comment]
@raych1 
I am getting a breaking change caused by sending a new property nextLink.
This should not be considered as  breaking change as the user behavior will not be changing and the user is additionally provided with a link to retrieve more results.
Can this be considered ?

Also not sure why automation-sdk is failing, will suppressing the breaking change fix it ?

https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Breaking%20changes%20guidelines.md#adding-read-only-field-to-response

+ @pilor for visibility.

**Comment by sandipsh** [comment]
>     "/{scope}/providers/Microsoft.PolicyInsights/policyEvents/$metadata": {

We removed it from policyStates recently. Please check if it should be removed from events too.

---
Refers to: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:603 in 1c9ee7f. [](commit_id = 1c9ee7f85f8e9bc82337fa4fdc795ea04c2f411d, deletion_comment = False)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyEvents_QueryManagementGroupScope.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/examples/PolicyEvents_QueryManagementGroupScope.json:14]: >value [](start = 9, length = 5)

Please also add "complianceState" field to the response. Same in all events json for this version.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/examples/PolicyEvents_QuerySubscriptionScope.json
Comment (sandipsh) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/examples/PolicyEvents_QuerySubscriptionScope.json:13]: >value [](start = 9, length = 5)

Please also add "complianceState" field to the response. Same in all events json for this version.

**Comment by sandipsh** [comment]
>       "description": "The $expand query parameter. For example, to expand components use $expand=components",

might want to mention that results won't be paginated for this expand option?

---
Refers to: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:1068 in 1c9ee7f. [](commit_id = 1c9ee7f85f8e9bc82337fa4fdc795ea04c2f411d, deletion_comment = False)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/examples/PolicyEvents_QuerySubscriptionScope.json
Comment (najagasi) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/examples/PolicyEvents_QuerySubscriptionScope.json:13]: hmm this should have already been there.
I will update add it where its missing

**Comment by raych1** [comment]
> @raych1
> I am getting a breaking change caused by sending a new property nextLink.
> This should not be considered as breaking change as the user behavior will not be changing and the user is additionally provided with a link to retrieve more results.
> Can this be considered ?
> 
> Also not sure why automation-sdk is failing, will suppressing the breaking change fix it ?
> 
> https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Breaking%20changes%20guidelines.md#adding-read-only-field-to-response
> 
> * @pilor for visibility.

Breaking change cannot be suppressed and needs to be forcibly merged if all other issue are resolved. For SDK Net failure, it can be ignored.

**Review by sandipsh — APPROVED** [review]
:shipit:

**Comment by najagasi** [comment]
@raych1 this PR is ready to merge from my side, have described the breaking change error that can be ignored in the above comment.

**Comment by raych1** [comment]
@najagasi , it needs ARM sign off.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (majastrz) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:2]: >swagger [](start = 3, length = 7)

Is this a completely new API or is there an older previously approved version that can be compared with this one?

Comment (majastrz) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:2]: Or are you backfilling a missing swagger?

---
In reply to: [417722565](https://github.com/Azure/azure-rest-api-specs/pull/7909#discussion_r417722565) [](ancestors = 417722565)

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (majastrz) [inline specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:618]: >nextLink [](start = 16, length = 8)

You're missing a lot of x-ms-pageable declarations in this file. (The easiest way to find them is to manually diff 2018-07-01-preview version of this file with this one.)

**Review by majastrz — CHANGES_REQUESTED** [review]
I added some comments. Please take a look.

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json
Comment (najagasi) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json:642]: Thank you had missed that, have added it now

File: specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json
Comment (najagasi) [reply specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyEvents.json:2]: these 2018-07-01-preview and 2019-10-01 both are older approved versions but they did not have paging enabled, my change in PR is to add x-ms-pageable and examples for that.

**Comment by najagasi** [comment]
@majastrz I have addressed your comment.
Please let me know if other changes are required 

thank you

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by najagasi** [comment]
@raych1 We have received ARM signoff and branch has the latest commits.
Please let me know If anything else is needed.

thank you

**Comment by raych1** [comment]
@akning-ms , the breaking change is adding a readonly property `nextlink`. Could you help force merge this PR?

**Review by raych1 — APPROVED** [review]

**Pull request merged**
